src/iptax/ai/provider.py

```
@@ -0,0 +1,238 @@
+            Tuple of (model_name, api_params_dict)
+
+        Raises:
+            AIProviderError: If configuration is invalid or credentials are missing
+        """
+        match self.config:
+            case GeminiProviderConfig():
+                return self._build_gemini_params()
+            case VertexAIProviderConfig():
+                return self._build_vertex_params()
+            case _:
+                raise AIProviderError(f"Unknown provider type: {self.config.provider}")
```

Ruff (0.14.7)

Avoid specifying long messages outside the exception class

(TRY003)